src/molecules/tables/index.tsx

```diff
@@ -517,47 +561,12 @@ export default function DataTable<TData, TValue>({
                   setFilteredColumns={setFilteredColumns}
                 />
               ))}
-            {detailedFilter?.filter(
-              (column) =>
-                filteredColumns?.findIndex((f) => f.name === column.name) === -1
-            )?.length > 0 && (
-              <DropdownMenu>
-                <DropdownMenuTrigger asChild>
-                  <Button
-                    disabled={isLoading}
-                    variant="outline"
-                    className="border px-3 py-1 border-gray-300 rounded-full text-xs mr-2 h-auto"
-                  >
-                    Filter <PlusIcon className="ml-2 h-4 w-4" />
-                  </Button>
-                </DropdownMenuTrigger>
-                <DropdownMenuContent align="end">
-                  {detailedFilter
-                    .filter(
-                      (column) =>
-                        filteredColumns.findIndex(
-                          (f) => f.name === column.name
-                        ) === -1
-                    )
-                    .map((column) => (
-                      <DropdownMenuItem
-                        key={column.name}
-                        className="capitalize"
-                        onClick={() =>
-                          setFilteredColumns((old) => [...old, column])
-                        }
-                      >
-                        {column.displayName}
-                      </DropdownMenuItem>
-                    ))}
-                </DropdownMenuContent>
-              </DropdownMenu>
-            )}
+            {getNonSelectedFilters().length > 0 && filterButton(detailedFilter)}
```

Comment:
Instread of this use it like <FilterButton detailedFilter={detailedFilter} />

This provides more performance because this way react will know that this is a react component. Also it let you use react life cycle methods.